src/main/java/org/isf/medicalstockward/manager/MovWardBrowserManager.java

```diff
@@ -295,4 +296,55 @@ public int compare(MovementForPrint o1, MovementForPrint o2) {
 		}
 	}
 
+	/* * 
+	 * Delete the MovementWard for the specified criteria.
+	 *
+	 * @param movWardToDelete - the movement ward
+	 * @throws OHServiceException if an error occurs retrieving the medical.
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void deleteLastMovementWard(MovementWard movWardToDelete) throws OHServiceException {
+		MovementWard lastMovementWard = ioOperations.getLastMovementWard(movWardToDelete.getWard());
+		if (lastMovementWard.getCode() != movWardToDelete.getCode()) {
+			throw new OHDataValidationException(
+							new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.onlythelastmovementcanbedeleted.msg")));
+		}
```

Comment:
Please extract variables when possible

```suggestion
		}
		Ward wardTo = movWardToDelete.getWardTo();
		Medical medical = movWardToDelete.getMedical();
		Lot lot = movWardToDelete.getLot();
```